mubench.reviewsite/src/Controllers/ReviewsController.php

```diff
@@ -41,10 +41,24 @@ public function getReview(Request $request, Response $response, array $args)
 
         $runs = RunsController::getRuns($detector, $experiment, $ex2_review_size);
 
-        $all_misuses = $this->collectAllMisuses($runs);
+        $limited_all_misuses = $this->collectAllMisuses($runs);
 
         list($previous_misuse, $next_misuse, $next_reviewable_misuse, $misuse) =
-            $this->determineNavigationTargets($all_misuses, $project_muid, $version_muid, $misuse_muid, $reviewer);
+            $this->determineNavigationTargets($limited_all_misuses, $project_muid, $version_muid, $misuse_muid, $reviewer);
+
+        if(!$misuse){
+            $runs = Run::of($detector)->in($experiment)->orderBy('project_muid')->orderBy('version_muid')->get();
+
+            $all_misuses = $this->collectAllMisuses($runs);
+
+            list($previous_misuse, $next_misuse, $next_reviewable_misuse, $misuse) =
+                $this->determineNavigationTargets($all_misuses, $project_muid, $version_muid, $misuse_muid, $reviewer);
+
+            $next_misuse = $all_misuses->first();
+            if($limited_all_misuses->where('id', $next_reviewable_misuse->id)->isEmpty()){
+                $next_reviewable_misuse = null;
+            }
+        }
```

Comment:
Try to reimplement `determineNavigationTargets` by starting from the current misuse and walking forward/backward to determine the targets.